src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. This would make your code more readable. For instance, you could have something like `List<? extends Machine> machines = someProducer.get();` and then return a new ArrayList of these machines.